rust/lancedb/src/database/listing.rs

```diff
@@ -322,6 +322,36 @@ impl ListingDatabase {
 
         Ok(uri)
     }
+
+    async fn drop_tables(&self, names: Vec<String>) -> Result<()> {
+        let object_store_params = ObjectStoreParams {
+            storage_options: Some(self.storage_options.clone()),
+            ..Default::default()
+        };
+        let mut uri = self.uri.clone();
+        if let Some(query_string) = &self.query_string {
+            uri.push_str(&format!("?{}", query_string));
+        }
+        let commit_handler = commit_handler_from_url(&uri, &Some(object_store_params)).await?;
+        for name in names {
+            let dir_name = format!("{}.{}", name, LANCE_EXTENSION);
+            let full_path = self.base_path.child(dir_name.clone());
+            self.object_store
+                .remove_dir_all(full_path.clone())
+                .await
+                .map_err(|err| match err {
+                    // this error is not lance::Error::DatasetNotFound, as the method
+                    // `remove_dir_all` may be used to remove something not be a dataset
+                    lance::Error::NotFound { .. } => Error::TableNotFound {
+                        name: name.to_owned(),
+                    },
+                    _ => Error::from(err),
+                })?;
+
+            commit_handler.delete(&full_path).await?;
```

Comment:
is the ordering right?

Reply:
It does seem a little backwards.  The default delete impl is a no-op.  The only commit handler with a custom delete handler is dynamodb and it drops the table.  So the only thing this could be doing is deleting records from dynamodb.  I think it would be safer to do this _before_ deleting the files.

If the file op succeeds but the dynamo op fails we will be in a weird corrupt state with failures.
If the dynamo op succeeds but the file op fails we will have wasted files in S3

Neither are great but the latter seems more harmless and could be manually repaired (by deleting the data).

Reply:
thanks, will update.